File: app/src/main/java/nu/huw/clarity/model/Attachment.java
package nu.huw.clarity.model;

import android.os.Parcel;
import android.os.Parcelable;
import java.io.File;

/**
 * Attachments.
 */
public class Attachment extends Base {

  public static final Parcelable.Creator<Attachment> CREATOR =
      new Parcelable.Creator<Attachment>() {
        public Attachment createFromParcel(Parcel in) {

          return new Attachment(in);
        }

        public Attachment[] newArray(int size) {

          return new Attachment[size];
        }
      };
  public String name;
  public String parentID;
  public File file;

  public Attachment() {
  }

  protected Attachment(Parcel in) {

    super(in);
    name = in.readString();
    parentID = in.readString();
    file = new File(in.readString());
  }

  @Override
  public void writeToParcel(Parcel out, int flags) {

    super.writeToParcel(out, flags);
    out.writeString(name);
    out.writeString(parentID);
    out.writeString(file.getPath());
  }
}

File: app/src/main/java/nu/huw/clarity/model/Attachment.kt
package nu.huw.clarity.model

import android.arch.persistence.room.Entity
import android.arch.persistence.room.TypeConverters
import nu.huw.clarity.db.Converters
import org.threeten.bp.LocalDateTime
import java.io.File

@Entity
@TypeConverters(Converters::class)
class Attachment(id: ID = ID(), dateAdded: LocalDateTime = LocalDateTime.now(), dateModified: LocalDateTime = LocalDateTime.now(), var name: String = "", var parentID: ID = ID(), var file: File = File("")) : Base(id, dateAdded, dateModified)